### PR TITLE
Add specs for invalid base argument in Kernel.Integer

### PR DESCRIPTION
**Pull request by herwinw**
I'm not sure if we're doing 3.3 changes already? Especially since it's not running in the CI yet

### CODE REVIEW

**Review by eregon — APPROVED** [review]
Yes it's fine, please make sure the new spec does pass on ruby-master or the latest preview though.

**Comment by herwinw** [comment]
Yes, I tried them locally with Ruby 3.2 en 3.3-preview3.

**Comment by eregon** [comment]
FYI it ends being tested in CRuby's CI after the monthly sync.
It doesn't make sense to test a preview or ruby-dev here as there are extra changes to specs in CRuby when behavior changes and that's not immediately sync'd here to it'd just fail a good part of the time for no good reason.

File: core/kernel/Integer_spec.rb [hunk 1]
```diff
@@ -586,6 +586,21 @@
...
+    end
+  end
+
+  ruby_version_is '3.3' do
+    it "raises a TypeError if it is not an integer and does not respond to #to_i" do
+      -> {
+        Integer("777", "8")
+      }.should raise_error(TypeError, "no implicit conversion of String into Integer")
+    end
+  end
```
Comment (andrykonchin) [inline core/kernel/Integer_spec.rb:602]: It seems Ruby calls `#to_int`, not `#to_i`

**Pull request merged**